src/components/DraggableLineChart/DraggableLineChartD3.ts

```diff
@@ -0,0 +1,186 @@
+import * as d3Axis from 'd3-axis';
+import * as d3Shape from 'd3-shape';
+import * as d3Drag from 'd3-drag';
+import * as d3Selection from 'd3-selection';
+import * as d3Transition from 'd3-transition';
+import { d3Scale } from '../../index';
+import _ from 'lodash';
+import * as d3Array from 'd3-array';
+
+interface IDraggableLineChartParams {
+	margin: {
+		top: number;
+		right: number;
+		bottom: number;
+		left: number;
+	};
+	height: number;
+	width: number;
+	data: Array<{ [key: string]: Date | string | number | undefined }>;
+	onDragEnd?: (d: any) => any;
+	cx: (d: any) => void;
+}
+
+class DraggableLineChartD3 {
+	selection: any;
+	params: IDraggableLineChartParams = {
+		margin: {
+			top: 10,
+			right: 80,
+			bottom: 65,
+			left: 80,
+		},
+		height: 300,
+		width: 1000,
+		onDragEnd: _.noop,
+		data: [],
+		cx: _.noop,
+	};
+	xScale: any;
```

Comment:
There is a lot of use of any. Let's either remove this or turn to jsx. TSX is just interfaces, so if you use any, we are not using the power of tsx

Reply:
Agree that 'any' kinda defeats the purpose of TypeScript but we've had difficulties making d3 play nicely with TS. Take a look at Lines.tsx and Points.tsx for some examples of how we handled xScale and yScale with TS. You can probably explicitly identify a type for these scales.